### ISSUE TITLE
MD013: long markdown link titles resulting in unfixable diagnostics

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#452]
### rumdl version

rumdl 0.1.25

### Installation method

Pre-built binary

### Bug description

a markdown link `()[]` with a long title results in a violation of MD013 which is not neither fixable with `rumdl fmt` nor with any kind of manual line splitting.

### Steps to reproduce

```markdown
## Case 1
- [some article looooooong title with a lot of words to be written for demonstration purposes](https://stackoverflow.blog/2020/11/25/how-to-write-an-effective-developer-resume-advice-from-a-hiring-manager/)

## Case 2
- [some article looooooong title with a lot of words to be written for 
  demonstration purposes](https://stackoverflow.blog/2020/11/25/how-to-write-an-effective-developer-resume-advice-from-a-hiring-manager/)
```

both cases result in a violation of MD013, splitting the markdown link title across two lines (the second case) does not resolve the issue.

### Expected behavior

- no rule violation, at least for the second case
- ideally, `rumdl fmt` should turn case 1 into case 2.

### Actual behavior

- rule violations for both cases
- `rumdl fmt` does fix case 1 into case 2.

### Configuration

```toml
[global]
line-length = 80
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.26 @chrisgrieser. Lines that only contain a link or image are now exempt from length checks in non-strict mode since there's no way to shorten them.
